source/Octopus.Tentacle.Contracts/Capabilities/BackwardsCompatibleCapabilitiesV2Decorator.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Generic;
+using Halibut;
+using Halibut.Exceptions;
+
+namespace Octopus.Tentacle.Contracts.Capabilities
+{
+    public class BackwardsCompatibleCapabilitiesV2Decorator : ICapabilitiesServiceV2
+    {
+        private readonly ICapabilitiesServiceV2 inner;
+
+        public BackwardsCompatibleCapabilitiesV2Decorator(ICapabilitiesServiceV2 inner)
+        {
+            this.inner = inner;
+        }
+
+        public CapabilitiesResponseV2 GetCapabilities()
+        {
+            try
+            {
+                return inner.GetCapabilities();
+            }
+            catch (NoMatchingServiceOrMethodHalibutClientException)
+            {
+                return new CapabilitiesResponseV2(new List<string>() {nameof(IScriptService), nameof(IFileTransferService)});
+            }
+            catch (HalibutClientException e) when 
+                (
+                    ExceptionLooksLikeTheServiceWasNotFound(e))
+            {
+                return new CapabilitiesResponseV2(new List<string>() {nameof(IScriptService), nameof(IFileTransferService)});
+            }
+        }
+
+        private static bool ExceptionLooksLikeTheServiceWasNotFound(HalibutClientException e)
```

Comment:
@nathanwoctopusdeploy it is worth re-reviewing this section which adds support for older tentacles.